### DISCUSSION TITLE
Is there an example to train a GCN semisupervised with multiple graphs

### DISCUSSION THREAD

**Discussion by harirRzi** [pyg-team/pytorch_geometric#9148] [Q&A]
Hello, I'm working with multiple graphs where each node is assigned a label of either 0 or 1. I'm interested in implementing semi-supervised learning, where I train a Graph Convolutional Network (GCN) using only a subset of labeled nodes in each graph.
The approach I'm considering involves iteratively selecting a portion of nodes in each graph and training the GCN to predict the labels of the remaining unlabeled nodes. I plan to repeat this process across multiple graphs until approximately 70% of the dataset has been utilized for training in this manner. Once the model is trained, I intend to evaluate its performance on test data to assess its ability to accurately predict node labels.
Do you have any examples or resources demonstrating how to implement this semi-supervised learning approach using GCNs?

**Comment by rusty1s** [comment]
This is similar to our `examples/gcn.py` file, but you would need to iterate over your graphs in batches via `DataLoader`. At best, each graph holds a `train_mask` that hold the nodes within each graph you want to use for training. Then, you can do something as follows:
```python
for batch in loader:
    batch = batch.to(device)
    out = model(batch.x, batch.edge_index)
    loss = loss_fn(out[batch.train_mask], batch.y[batch.train_mask])
    loss.backward()
    optimizer.step()
